### PR TITLE
fix test failure

### LINKED ISSUE #1046: [BUG] DataTypePKSuite failed because of Duplicate entry
**Describe the bug**

DataTypePKSuite failed because of Duplicate entry

https://internal.pingcap.net/idc-jenkins/blue/rest/organizations/jenkins/pipelines/tispark_regression_test_daily/runs/47/nodes/803/steps/1107/log/?start=0

```
generate data is enabled and seed is 1566241122598
```

```
DataTypePKSuite:
[2019-08-19T18:59:43.061Z] org.apache.spark.sql.types.pk.DataTypePKSuite *** ABORTED ***
[2019-08-19T18:59:43.061Z]   java.lang.RuntimeException: Unable to load a Suite class that was discovered in the runpath: org.apache.spark.sql.types.pk.DataTypePKSuite
[2019-08-19T18:59:43.061Z]   at org.scalatest.tools.DiscoverySuite$.getSuiteInstance(DiscoverySuite.scala:81)
[2019-08-19T18:59:43.061Z]   at org.scalatest.tools.DiscoverySuite$$anonfun$1.apply(DiscoverySuite.scala:38)
[2019-08-19T18:59:43.061Z]   at org.scalatest.tools.DiscoverySuite$$anonfun$1.apply(DiscoverySuite.scala:37)
[2019-08-19T18:59:43.061Z]   at scala.collection.TraversableLike$$anonfun$map$1.apply(TraversableLike.scala:234)
[2019-08-19T18:59:43.061Z]   at scala.collection.TraversableLike$$anonfun$map$1.apply(TraversableLike.scala:234)
[2019-08-19T18:59:43.061Z]   at scala.collection.Iterator$class.foreach(Iterator.scala:891)
[2019-08-19T18:59:43.061Z]   at scala.collection.AbstractIterator.foreach(Iterator.scala:1334)
[2019-08-19T18:59:43.061Z]   at scala.collection.IterableLike$class.foreach(IterableLike.scala:72)
[2019-08-19T18:59:43.062Z]   at scala.collection.AbstractIterable.foreach(Iterable.scala:54)
[2019-08-19T18:59:43.062Z]   at scala.collection.TraversableLike$class.map(TraversableLike.scala:234)
[2019-08-19T18:59:43.062Z]   ...
[2019-08-19T18:59:43.062Z]   Cause: com.mysql.jdbc.exceptions.jdbc4.MySQLIntegrityConstraintViolationException: Duplicate entry 'kaduyxbMFRfVxwfkYjMNNnMuzX' for key 'PRIMARY'
[2019-08-19T18:59:43.062Z]   at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
[2019-08-19T18:59:43.062Z]   at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
[2019-08-19T18:59:43.062Z]   at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
[2019-08-19T18:59:43.062Z]   at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
[2019-08-19T18:59:43.062Z]   at com.mysql.jdbc.Util.handleNewInstance(Util.java:425)
[2019-08-19T18:59:43.062Z]   at com.mysql.jdbc.Util.getInstance(Util.java:408)
[2019-08-19T18:59:43.062Z]   at com.mysql.jdbc.SQLError.createSQLException(SQLError.java:935)
[2019-08-19T18:59:43.062Z]   at com.mysql.jdbc.MysqlIO.checkErrorPacket(MysqlIO.java:3973)
[2019-08-19T18:59:43.062Z]   at com.mysql.jdbc.MysqlIO.checkErrorPacket(MysqlIO.java:3909)
[2019-08-19T18:59:43.062Z]   at com.mysql.jdbc.MysqlIO.sendCommand(MysqlIO.java:2527)
```

### PR DESCRIPTION
**Pull request by zhexuany**
### What problem does this PR solve? 
This PR closes  #1046, #1047, and #1048 
### What is changed and how it works?
We did not handle array[boolean] well in hash method.

### CODE REVIEW

**Comment by zhexuany** [comment]
@birdstorm @marsishandsome PTAL

File: core/src/test/scala/org/apache/spark/sql/test/generator/ColumnValueGenerator.scala [hunk 1]
```diff
@@ -106,7 +107,7 @@ case class ColumnValueGenerator(dataType: ReflectedDataType,
 
   private val specialBound: List[Any] = {
     val list: List[Any] = dataType match {
-      case BIT                                                                     => List(Array[Byte]())
+      case BIT                                                                     => List.empty[Array[Byte]]
```
Comment (birdstorm) [inline core/src/test/scala/org/apache/spark/sql/test/generator/ColumnValueGenerator.scala]: we may delete this case branch

File: core/src/test/scala/org/apache/spark/sql/test/generator/TestDataGenerator.scala [hunk 2]
```diff
@@ -303,6 +306,7 @@ object TestDataGenerator {
   def checkUnique(value: Any, set: mutable.Set[Any]): Boolean = {
     val hashedValue = hash(value)
     if (!set.apply(hashedValue)) {
+      logger.info(s"hashedValue is $hashedValue")
```
Comment (birdstorm) [inline core/src/test/scala/org/apache/spark/sql/test/generator/TestDataGenerator.scala]: remove logs

**Pull request merged**